scripts/unified_state_manager.py

```
@@ -0,0 +1,632 @@
+                    yield lock
+            except filelock.Timeout:
+                raise TimeoutError(f"Could not acquire lock within {timeout} seconds")
+        elif HAS_FCNTL:
+            # Unix/Linux fallback
+            lock_fd = os.open(str(self.lock_file), os.O_CREAT | os.O_WRONLY)
+            try:
+                # Try to acquire exclusive lock
+                fcntl.lockf(lock_fd, fcntl.LOCK_EX | fcntl.LOCK_NB)
+                yield lock_fd
+            except BlockingIOError:
+                raise TimeoutError(f"Could not acquire lock (file in use)")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)